app/routes/search.tsx

```diff
@@ -1,9 +1,44 @@
+import { LoaderFunction } from '@remix-run/node';
+import { useLoaderData } from '@remix-run/react';
+import { getEvents } from '~/models/events.server';
+
+export const loader: LoaderFunction = async ({ request }) => {
+  // Get the query string from the URL
+
+  const url = new URL(request.url);
+  const query = url.searchParams.get('q') as string;
```

Comment:
Instead of casting to string (lying to TypeScript), you may be able to cast the value to a string:

`url.searchParams.get('q').toString()` or check if the value is a string `typeof q === "string"` and only then proceed!

Reply:
100%. Let's check the type.